src/java.desktop/share/classes/sun/print/PathGraphics.java

```diff
@@ -934,23 +934,36 @@ boolean printedSimpleGlyphVector(GlyphVector g, float x, float y) {
 
         /* If we reach here we have mapped all the glyphs back
          * one-to-one to simple unicode chars that we know are in the font.
-         * We can call "drawChars" on each one of them in turn, setting
+         * We can call "drawString" on each one of them in turn, setting
          * the position based on the glyph positions.
          * There's typically overhead in this. If numGlyphs is 'large',
          * it may even be better to try printGlyphVector() in this case.
          * This may be less recoverable for apps, but sophisticated apps
          * should be able to recover the text from simple glyph vectors
          * and we can avoid penalising the more common case - although
          * this is already a minority case.
+         * If we do use "drawString" on each character, we need to use a
+         * font without translation transform, since the font translation
+         * transform will already be reflected in the glyph positions, and
+         * we do not want to apply the translation twice.
          */
         if (numGlyphs > 10 && printGlyphVector(g, x, y)) {
             return true;
         }
 
+        Font font2 = font;
+        if (font.isTransformed()) {
+            AffineTransform t = font.getTransform();
+            if ((t.getType() & AffineTransform.TYPE_TRANSLATION) != 0) {
+                t.setTransform(t.getScaleX(), t.getShearY(), t.getShearX(), t.getScaleY(), 0, 0);
+                font2 = font.deriveFont(t);
+            }
+        }
+
```

Comment:
The fix to PathGraphics.java looks good.

A new font object is created where you copy over all the transforms except translation from the original font object and use the newly created font object in drawString(). The translation transform is skipped since the drawString takes care of x and y positions.